Science/dust flux driver

klindsay28 opened the pull request:
corresponding POP marbl_dev_klindsay tag is
marbl_dev_klindsay_n06_marbl_dev_n07_cesm_pop_2_1_20160426

klindsay28 committed:
add dust_flux_source namelist variable, valid values='monthly-calendar,driver'

base dust_flux add_forcing_field on dust_flux_source

Testing:
with marbl_dev_klindsay_n03
ecosys_tavg.F90 from cesm_pop_2_1_20160413 included in testing, but not in this tag

aux_pop_MARBL: (baseline comparison to n01 tag)
   all tests pass, except nlcomp

aux_pop_MARBL (gnu): (baseline comparison to n00 tag)
   all tests pass, except nlcomp

Files Modified:
	modified:   src/marbl_interface_types.F90
	modified:   src/marbl_mod.F90
	modified:   src/marbl_share_mod.F90

klindsay28 committed:
Tag Summary: merge latest NCAR/MARBL branch tag to this branch

Testing:
with marbl_dev_klindsay_n04

aux_pop_MARBL: (baseline comparison to n03 tag)
   all tests pass, except nlcomp, which all fail

aux_pop_MARBL (gnu): (baseline comparison to n03 tag)
   all tests pass, except nlcomp, which all fail

Files Modified:
	modified:   .gitignore
	deleted:    ChangeLog
	modified:   src/marbl_ciso_mod.F90
	modified:   src/marbl_co2calc_mod.F90
	modified:   src/marbl_diagnostics_mod.F90
	modified:   src/marbl_interface.F90
	modified:   src/marbl_interface_types.F90
	modified:   src/marbl_internal_types.F90
	modified:   src/marbl_logging.F90
	modified:   src/marbl_mod.F90
	modified:   src/marbl_parms.F90
	modified:   src/marbl_restore_mod.F90
	modified:   src/marbl_share_mod.F90
	modified:   src/marbl_sizes.F90

klindsay28 committed:
Tag Summary: enable iron flux to be derived from driver provided dust and black carbon

add black_carbon_flux_id to marbl_surface_forcing_indexing_type
add iron_flux_source namelist variable, valid values='monthly-calendar,driver-derived'
add namelist variables related to iron_flux_source='driver-derived'
   iron_frac_in_dust, iron_frac_in_bc
base iron_flux, black_carbon_flux add_forcing_file calls on iron_flux_source
add checking of valid values to following namelist variables
   dust_flux_source, iron_flux_source, ndep_data_type
corect subname values in files already being edited

Testing:
with marbl_dev_klindsay_n06

aux_pop_MARBL (intel and gnu): (baseline comparison to n04 tag)
   all tests pass, with the following exceptions
      some nlcomp tests fail because of added namelist variables
      cice_ecosys_dust_flux_driver fails baseline compare because of changed test defn

aux_pop_MARBL on hobart (nag) will be done after commit
   lack of subsequent tags indicates success

Files Modified:
	modified:   marbl_diagnostics_mod.F90
	modified:   marbl_interface_types.F90
	modified:   marbl_mod.F90
	modified:   marbl_parms.F90
	modified:   marbl_share_mod.F90

klindsay28 commented:
aux_pop_MARBL test suite on hobart with nag passes with following execptions
   SMS_Ld_D test fails
   some nlcomp tests fail because added namelist variables
(the baseline is to the previous tags and commits on the respective branches, not to NCAR/master

mnlevy1981 commented:
The code looks good, but should the following be addressed before the pull request is accepted?

```
! FIXME : check coefficient values in formula here
```

https://github.com/NCAR/MARBL/pull/61/files?w=1#diff-a1264867a6cdb3406ca4dadf7e51bfeaR2986

I still need to look at changes to `ecosys_driver.F90` in the POP repository.

mnlevy1981 commented:
Another question that came up when I realized you were adding new MARBL namelist variables (i.e. while looking at the driver code): you initialize `iron_frac_in_dust` with

```
iron_frac_in_dust            = 0.035_r8 * 0.01_r8
```

https://github.com/NCAR/MARBL/pull/61/files?w=1#diff-a1264867a6cdb3406ca4dadf7e51bfeaR498

Is there a reason you do this multiplication explicitly? From the comments surrounding the definition of the variable, it seems like it is unitless... so it's not a unit conversion factor (as far as I can tell). Initializing with

```
iron_frac_in_dust            = 3.5e-4_r8
```

seems "cleaner" to me, unless there is a good reason to define it as is.

My only other comment from the driver (at the moment) is that you have another FIXME that should be addressed before merging (`ecosys_driver.F90`, line 1664):

```
! FIXME : is this scaling correct if dust_flux_source = 'driver' ?
```

klindsay28 committed:
Tag Summary: clean up comments, log_messages, and unit metadata

Testing: verified that code builds with ifort and gfortran on yellowstone

Files Modified:
	modified:   src/marbl_diagnostics_mod.F90
	modified:   src/marbl_mod.F90
